src/capymoa/regressor/_soknl.py

```diff
@@ -1,17 +1,40 @@
 # Library imports
-from typing import Optional, Union
+
+from ._soknl_base_tree import SOKNLBT
+from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
 
 from capymoa.base import (
     MOARegressor, _extract_moa_learner_CLI
 )
 
-from capymoa.splitcriteria import SplitCriterion, _split_criterion_to_cli_str
-from capymoa.stream._stream import Schema
-from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
-from ._soknl_base_tree import SOKNLBT
-
 
 class SOKNL(MOARegressor):
+    """Self-Optimising K-Nearest Leaves (SOKNL) Implementation.
+
+    SOKNL is an extension to AdaptiveRandomForestRegressor.
```

Comment:
SOKNL extends the AdaptiveRandomForestRegressor by limiting the number of base trees (or tree leaves) involved in predicting a given instance. This approach overrides the aggregation strategy used for voting, leading to more accurate predictions in general.